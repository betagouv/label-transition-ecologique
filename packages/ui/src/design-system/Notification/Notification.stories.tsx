import {Meta, StoryObj} from '@storybook/react';
import {Notification, NotificationSize, NotificationVariant} from '.';

const meta: Meta<typeof Notification> = {
  component: Notification,
};

export default meta;

type Story = StoryObj<typeof Notification>;

export const NombreUniquement: Story = {
  args: {number: 10},
};

export const IconeUniquement: Story = {
  args: {icon: 'lock-fill'},
};

export const NombreEtIcone: Story = {
  args: {number: 10, icon: 'lock-fill'},
};

export const Variantes: Story = {
  parameters: {},
  render: () => (
    <div
      className="grid gap-5 items-center bg-grey-2 p-10"
      style={{gridTemplateColumns: 'repeat(5,fit-content(0))'}}
    >
      {['md', 'sm', 'xs'].map(size => (
        <div key={size} className="flex flex-col gap-5">
          {['default', 'warning', 'info', 'error'].map(variant => (
            <Notification
              number={10}
              icon="lock-fill"
              variant={variant as NotificationVariant}
              size={size as NotificationSize}
            />
          ))}
        </div>
      ))}
    </div>
  ),
};
