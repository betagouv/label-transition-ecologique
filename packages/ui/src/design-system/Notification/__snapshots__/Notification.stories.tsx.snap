// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design-system/Notification IconeUniquement smoke-test 1`] = `
<div class="text-primary fill-primary bg-primary-3 text-md inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
  <span class="ri-lock-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem]">
  </span>
</div>
`;

exports[`design-system/Notification NombreEtIcone smoke-test 1`] = `
<div class="text-primary fill-primary bg-primary-3 text-md inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
  <span class="font-extrabold leading-none">
    10
  </span>
  <span class="ri-lock-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem]">
  </span>
</div>
`;

exports[`design-system/Notification NombreUniquement smoke-test 1`] = `
<div class="text-primary fill-primary bg-primary-3 text-md inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
  <span class="font-extrabold leading-none">
    10
  </span>
</div>
`;

exports[`design-system/Notification Variantes smoke-test 1`] = `
<div class="grid gap-5 items-center bg-grey-2 p-10"
     style="grid-template-columns: repeat(5, fit-content(0px));"
>
  <div class="flex flex-col gap-5">
    <div class="text-primary fill-primary bg-primary-3 text-md inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-warning-1 text-md inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-info-1 text-md inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-error-1 text-md inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-5 w-5 text-[1.25rem] leading-[1.25rem]">
      </span>
    </div>
  </div>
  <div class="flex flex-col gap-5">
    <div class="text-primary fill-primary bg-primary-3 text-sm inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-warning-1 text-sm inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-info-1 text-sm inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-error-1 text-sm inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-4 w-4 text-[1rem] leading-[1rem]">
      </span>
    </div>
  </div>
  <div class="flex flex-col gap-5">
    <div class="text-primary fill-primary bg-primary-3 text-xs inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-3.5 w-3.5 text-[0.875rem] leading-[0.875rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-warning-1 text-xs inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-3.5 w-3.5 text-[0.875rem] leading-[0.875rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-info-1 text-xs inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-3.5 w-3.5 text-[0.875rem] leading-[0.875rem]">
      </span>
    </div>
    <div class="text-white fill-white bg-error-1 text-xs inline-flex items-center gap-1 p-2 rounded-full border-2 border-grey-1 shadow">
      <span class="font-extrabold leading-none">
        10
      </span>
      <span class="ri-lock-fill font-normal text-center h-3.5 w-3.5 text-[0.875rem] leading-[0.875rem]">
      </span>
    </div>
  </div>
</div>
`;
